libdnf/rpm/solv/map.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "map.hpp"
+
+
+namespace libdnf::rpm::solv {
+
+
+SolvMap::SolvMap(int32_t size) {
+    map_init(&map, size);
+}
+
+
+SolvMap::SolvMap(const Map * other) {
+    map_init_clone(&map, other);
+}
+
+
+SolvMap::SolvMap(const SolvMap & other)
+    : SolvMap(other.get_map())
+{
+}
+
+
+SolvMap::~SolvMap() {
+    map_free(&map);
+}
+
+
+const Map * SolvMap::get_map() const {
+    return &map;
+}
+
+
+bool SolvMap::empty() const {
+    const unsigned char * byte = map.map;
+    const unsigned char * end = byte + map.size;
+
+    // iterate through the whole bitmap by moving the address
+    while (byte < end) {
+        if (*byte++) {
+            // return false if a non-zero bit was found
+            return false;
+        }
+    }
+    // all bits were zero, return true
+    return true;
+}
+
+
+std::size_t SolvMap::size() const {
+    unsigned char * byte = map.map;
+    unsigned char * end = byte + map.size;
+    std::size_t result = 0;
+
+    // iterate through the whole bitmap by moving the address
+    while (byte < end) {
+        // add number of bits in each byte
+        result += BIT_COUNT_LOOKUP[*byte++];
+    }
+    return result;
+}
+
+
+void SolvMap::clear() {
```

Comment:
I believe that it could be created as an inline. Also all following functions. Or did I overlooked something?

Reply:
I turned all methods into inlines.